lib/Platform.Posix.cs

```diff
@@ -16,11 +16,7 @@ public static class Posix
 		{
 			private const CallingConvention CCCdecl = CallingConvention.Cdecl;
 
-			private const string __Internal = "__Internal";
-
-			// private const string LibraryName = "libdl";
-
-			// public const string LibraryFileExtension = ".so";
+			private const string LibraryName = "libdl";
```

Comment:
I don't think there is a libdl on Mac OS X (see, e.g. https://stackoverflow.com/questions/20169660/where-is-libdl-so-on-mac-os-x). This must be done in a way dependent on the execution platform.

Reply:
I was afraid of that. I'll change it so it uses the __Internal library if running in Mono. It should probably be split into two classes, but I may just add it to this class for the time being.

Reply:
I see on "MacOSX libdl.dylib is a symlink to libSystem.dylib"...

However, this is trying just `libdl`, due to `DllImport["libdl"]`, which I wanted to circumvent using `MonoDllMapInsert`, which is basically saying that behind this "libdl" is for real "libdl.dylib"/"libdl.so".

Now look: I do believe you need to do `Platform.LoadUnmanagedLibrary(Platform.LibraryName)`, just AFTER `Platform.SetupImplementation(typeof(Platform))` in `static Platform()`.

If you don't know what I'm trying to say ;-) Tomorrow, I'm going to pull your request and try this approach!

Reply:
If you think you can fix it, give it a shot. I don't have an instance of OSX or Mono setup for testing.